backend/services/foodService.ts

```diff
@@ -0,0 +1,52 @@
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+export async function saveFoodData(
+  userId: string,
+  imageData: string,
+  foodData: any
+) {
+  try {
+    // Early return if no food data or food items
+    if (!foodData || !foodData.foodItems || foodData.foodItems.length === 0) {
+      console.log("No food items detected, skipping database save");
+      return null;
+    }
+
+    // First, ensure user exists (create if not)
+    const user = await prisma.user.upsert({
+      where: { clerkId: userId },
+      update: {},
+      create: {
```

Comment:
I don't think we need this. This is handled by the webhook handler which creates a user after a new user is created in Clerk. Of course, it's possible for the webhook handler to fail, but I don't think the foodService should be responsible for handling that specific case explicitly.